internal/stacks/stackruntime/plan_test.go

```diff
@@ -1705,6 +1705,266 @@ func TestPlanWithDeferredComponentForEach(t *testing.T) {
 	}
 }
 
+func TestPlanWithDeferredEmbeddedStackForEach(t *testing.T) {
```

Comment:
Oof, these tests are so brutal to read. What do you think about summarizing the intended behavior in a comment? Even just something like `// Unknown for_each defers a stack call, which then defers all its components and all of those components' resources.` It gives the next maintainer a bit more to work with than just the test func name.